students/rob_sanchez/Lesson 5/mailroom_pt3.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+import sys
+import re
+
+
+# List of donors and donation amounts
+donor_dict = {"Tom Cruise": [100, 200, 300],
+              "Michael Jordan": [1300],
+              "Katy Perry": [4500, 1500],
+              "Adam Sandler": [500, 2400]}
+
+
+def main():
+    # Menu options
+    options = {1: send_thank_you,
+               2: create_report,
+               3: send_letters,
+               4: sys.exit}
+    prompt = "\nChoose an action:\n"
+    menu_sel = ("\n1 - Send a Thank You\n2 - Create a Report\n"
+                "3 - Send letters to everyone\n4 - Quit\n")
+
+    # User selection
+    while True:
+        try:
+            user_selection = input(prompt + menu_sel)
+            options.get(int(user_selection))()
+        except ValueError:
+            print("\nPlease select a numeric value...")
+        except TypeError:
+            print("\nOption {} is invalid. Try again...".format(user_selection))
+
+
+# Sends a thank you email to the selected donor
+def send_thank_you():
+    # Holds a history of the donor's name and donation amount
+    history = {}
+
+    # Create list of current donors
+    name_set = set()
+    for donor, value in donor_dict.items():
+        name_set.add(donor)
```

Comment:
couple things here:
* since you're not using `value` var u should only loop over keys: `for name in donor_dict` should do
* using a set here doesn't really add value since names are unique (cannot have duplicate keys in dict)
* to actually get a list of names -> `names = list(donor_dict)` very simple 🙂